### PR TITLE
DEP-2770 Added logic to prevent repeated user surveys.

### PR DESCRIPTION
**Pull request by jameshforster**
# DEP-(StoryNumberHere)

## Checklist PR Raiser
 - [ ]  I've ensured code coverage has not decreased
 - [x]  I've dealt with any new compilation warnings
 - [x]  I've ensured the team's coding standards have been met (TBC)?
 - [ ]  If relevant, I've created corresponding app-config-XXX changes?
 - [ ]  If I've created new test data, I've ensured there is no real person's data (see https://confluence.tools.tax.service.gov.uk/display/CD/Test+Data+in+the+Open) 
 - [ ]  I've considered the impact my changes have on the UI/Journey tests (please do not break them)
-  [ ]  If relevant, I've added links to associated PRs

## Checklist PR Reviewer
 - [ ]  I've checked to ensure all relevant unit tests have been written
 - [ ]  I've checked to ensure the team's coding standards have been met
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  If I merge I will ensure I use squash and merge

### CODE REVIEW

File: app/assets/javascripts/controllers/CommonChatController.js [hunk 1]
```diff
@@ -380,21 +379,32 @@ export default class CommonChatController {
...
+		} else {
+			document.cookie = "surveyed=true";
+			if (escalated) {
+				this._sendPostChatSurveyWebchat(this.sdk).beginPostChatSurvey(webchatSurvey, automatonWebchat, timestamp);
+				this.container.showPage(new PostChatSurveyWebchat((page) => this.onPostChatSurveyWebchatSubmitted(page)));
+			} else {
+				this._sendPostChatSurveyDigitalAssistant(this.sdk).beginPostChatSurvey(digitalAssistantSurvey, automatonDA, timestamp);
+				this.container.showPage(new PostChatSurveyDigitalAssistant((page) => this.onPostChatSurveyDigitalAssistantSubmitted(page)));
+			}
+			window.GOVUKFrontend.initAll();
```
Comment (machinesleet) [inline app/assets/javascripts/controllers/CommonChatController.js:405]: perhaps a little comment about what this line does

File: app/assets/javascripts/controllers/CommonChatController.js [hunk 2]
```diff
@@ -239,9 +239,8 @@ export default class CommonChatController {
 
     closeChat() {
 
-        if (document.body.contains(document.getElementById("postChatSurveyWrapper"))) {
+		if (document.body.contains(document.getElementById("postChatSurveyWrapper"))) {
```
Comment (machinesleet) [inline app/assets/javascripts/controllers/CommonChatController.js:242]: something a bit funky going on with the tabs/spaces in this file

**Comment by machinesleet** [comment]
When does the cookie get reset?

File: app/assets/javascripts/controllers/CommonChatController.js (hunk 1 above)
Comment (tonymyers) [reply app/assets/javascripts/controllers/CommonChatController.js:405]: do we still need the console.log?

**Review by tonymyers — APPROVED** [review]

**Pull request merged**